lambda/src/new_account_support_case.py

```diff
@@ -0,0 +1,222 @@
+            return account_status["CreateAccountStatus"]["AccountId"]
+        if state == "FAILED":
+            LOG.error({"create_account_status_failure": account_status})
+            raise AccountCreationFailedError
+        LOG.info({"create_account_status_state": state})
+        time.sleep(5)
+
+
+def get_invite_account_id(event):
+    """Return account id for invite account events."""
+    return event["detail"]["requestParameters"]["target"]["id"]
+
+
+def get_account_id(event):
+    """Return account id for supported events."""
+    event_name = event["detail"]["eventName"]
+    get_account_id_strategy = {
+        "CreateAccount": get_new_account_id,
+        "CreateGovCloudAccount": get_new_account_id,
+        "InviteAccountToOrganization": get_invite_account_id,
+    }
+    try:
+        account_id = get_account_id_strategy[event_name](event)
+    except (botocore.exceptions.ClientError, AccountCreationFailedError) as err:
+        raise AccountCreationFailedError(err) from err
+    return account_id
+
+
+### Classes and functions specific to creating the support case.
+
+
+class SupportCaseInvalidArgumentsError(Exception):
+    """Invalid arguments were used to create a support case."""
+
+
+class SupportCaseError(Exception):
+    """Error creating or extracting details for support case."""
+
+
+def main(new_account_id, company_name, cc_list):
+    """Create an Enterprise support case and extract the display ID."""
+    # Prepare some of the fields needed for the create_case() call.
+    case_subject = (
+        f"Add new account to {company_name} Enterprise support: {new_account_id}"
+    )
+    case_communication_body = (
+        f"Hi AWS, please add this account to Enterprise support and add tax "
+        f"exemption (see the Tax Exemption letter under the payer account): "
+        f"{new_account_id}"
+    )
+
+    # Create the Enterprise support case.
+    client = boto3.client("support")
+    try:
+        response = client.create_case(
+            subject=case_subject,
+            severityCode="low",
+            categoryCode="other-account-issues",
+            serviceCode="customer-account",
+            communicationBody=case_communication_body,
+            ccEmailAddresses=[cc_list],
+            language="en",
+            issueType="customer-service",
+        )
+    except botocore.exceptions.ClientError as err:
+        raise SupportCaseError(err) from err
+
+    # Extract case ID from response.
+    case_id = response["caseId"]
+    if not case_id:
+        raise SupportCaseError("Missing case_id in create_case() response")
+
+    # Use the case ID to obtain further details from the case and in
+    # particular, the display ID.
+    try:
+        case = client.describe_cases(caseIdList=[case_id])
+    except botocore.exceptions.ClientError as err:
+        raise SupportCaseError(err) from err
+
+    if "cases" in case and len(case["cases"]) > 0 and "displayId" in case["cases"][0]:
```

Comment:
This seems like maybe overthinking it. Can we use a try/except instead?

```
try:
    dislay_id = case["cases"][0]["displayId"]
except KeyError as err:
    raise SupportCaseError("Unable to extract 'displayId' from case info") from err
```

Reply:
Done